### PR TITLE
[AppConfig] Run pyright

### PR DESCRIPTION
**Pull request by YalinLi0312**
Not option-in pyright, because there are 4 errors don't know how to fix, but they are gone when running next-pyright.
There are 3 errors in next-pyright result, I fixed one of them here, the rest 2 are from generated code.

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def __init__(self, base_url: str, credential: TokenCredential, **kwargs: Any) ->
         if pipeline is None:
             aad_mode = not isinstance(credential, AppConfigConnectionStringCredential)
             pipeline = self._create_appconfig_pipeline(
-                credential=credential, aad_mode=aad_mode, base_url=base_url, **kwargs
+                credential=credential, credential_scopes=self._credential_scopes, aad_mode=aad_mode, **kwargs
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:89]: Why we don't need base_url any more?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 2]
```diff
@@ -59,15 +59,14 @@ class ConfigurationSetting(Model):
...
-    content_type = None
 
     def __init__(self, **kwargs: Any) -> None:
         super(ConfigurationSetting, self).__init__(**kwargs)
         self.key = kwargs.get("key", None)
         self.label = kwargs.get("label", None)
         self.value = kwargs.get("value", None)
         self.etag = kwargs.get("etag", None)
-        self.content_type = kwargs.get("content_type", self.content_type)
+        self.content_type = kwargs.get("content_type", None)
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:69]: Why defaults to None?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 1 above)
Comment (lmazuel) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:89]: Agree @xiangyan99, improving typing shouldn't change the code, unless it was a bug, and in that case we should create an issue number explaining what we discovered, so we can explain it in the changelog.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 3]
```diff
@@ -359,7 +358,7 @@ def set_configuration_setting(
 
         try:
             key_value_set = self._impl.put_key_value(
-                entity=key_value,
+                entity=key_value,  # pyright: ignore
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:361]: Same, please comment

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
 if sys.version_info >= (3, 8):
     from typing import Literal  # pylint: disable=no-name-in-module, ungrouped-imports
 else:
-    from typing_extensions import Literal  # type: ignore  # pylint: disable=ungrouped-imports
+    from typing_extensions import Literal  # pylint: disable=ungrouped-imports
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:22]: Just import from typing_extensions all the time, there are no perf penalities for doing so, compared to the if/else here

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 5]
```diff
@@ -59,15 +59,14 @@ class ConfigurationSetting(Model):
     }
 
     kind = "Generic"
-    content_type = None
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:62]: It's technically a breaking change if this model was public

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 6]
```diff
@@ -40,7 +40,7 @@
 if sys.version_info >= (3, 8):
     from typing import Literal  # pylint: disable=no-name-in-module, ungrouped-imports
 else:
-    from typing_extensions import Literal  # type: ignore  # pylint: disable=ungrouped-imports
+    from typing_extensions import Literal  # pylint: disable=ungrouped-imports
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:43]: Just import from typing_extensions all the time, there are no perf penalities for doing so, compared to the if/else here

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 7]
```diff
@@ -270,12 +270,12 @@ async def add_configuration_setting(
...
         """
         key_value = configuration_setting._to_generated()
-        custom_headers = CaseInsensitiveDict(kwargs.get("headers"))  # type: Mapping[str, Any]
+        custom_headers: Mapping[str, Any] = CaseInsensitiveDict(kwargs.get("headers"))
         error_map = {412: ResourceExistsError}
 
         try:
             key_value_added = await self._impl.put_key_value(
-                entity=key_value,
+                entity=key_value,  # pyright: ignore
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:278]: Same, please explain why

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:69]: I don't change the default value, the original value of self.content_type is None.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 8]
```diff
@@ -290,11 +289,11 @@ def add_configuration_setting(self, configuration_setting: ConfigurationSetting,
             added_config_setting = client.add_configuration_setting(config_setting)
         """
         key_value = configuration_setting._to_generated()
-        custom_headers = CaseInsensitiveDict(kwargs.get("headers"))  # type: Mapping[str, Any]
+        custom_headers: Mapping[str, Any] = CaseInsensitiveDict(kwargs.get("headers"))
         error_map = {412: ResourceExistsError}
         try:
             key_value_added = self._impl.put_key_value(
-                entity=key_value,
+                entity=key_value,  # pyright: ignore
```
[the remark this answers is not in the document]
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: Removed all "pyright: ignore" and keep "pyright=False" instead.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py (hunk 4 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py]: Updated

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py (hunk 5 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py]: Added back

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py (hunk 6 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py]: Updated

**Review by lmazuel — APPROVED** [review]

**Pull request merged**